File: cwt-spring/src/test/java/com/cwtsite/cwt/domain/user/view/model/UserStatsDtoTest.java
package com.cwtsite.cwt.domain.user.view.model;

import org.junit.Assert;
import org.junit.Test;

public class UserStatsDtoTest {

    @Test
    public void locRound() {
        Assert.assertEquals(UserStatsDto.locRound(5, 7), "Champion");
        Assert.assertEquals(UserStatsDto.locRound(5, 6), "Runner-up");
        Assert.assertEquals(UserStatsDto.locRound(5, 5), "Third");
        Assert.assertEquals(UserStatsDto.locRound(5, 4), "Semifinal");
        Assert.assertEquals(UserStatsDto.locRound(5, 3), "Quarterfinal");
        Assert.assertEquals(UserStatsDto.locRound(5, 2), "Last Sixteen");
        Assert.assertEquals(UserStatsDto.locRound(5, 1), "Group");
        Assert.assertEquals(UserStatsDto.locRound(5, 0), "Not attended");
        Assert.assertEquals(UserStatsDto.locRound(6, 1), "Group");
        Assert.assertEquals(UserStatsDto.locRound(6, 3), "Last Sixteen");
        Assert.assertEquals(UserStatsDto.locRound(6, 2), "Last 32");
        Assert.assertEquals(UserStatsDto.locRound(7, 2), "Last 64");
        Assert.assertEquals(UserStatsDto.locRound(7, 3), "Last 32");
    }
}

File: cwt-spring/src/test/java/com/cwtsite/cwt/domain/user/view/model/UserStatsDtoTest.kt
package com.cwtsite.cwt.domain.user.view.model

import org.assertj.core.api.Assertions
import org.junit.Test

class UserStatsDtoTest {

    @Test
    fun toDtos() {
        // 8 players in playoffs
        Assertions
                .assertThat(UserStatsDto.toDtos("[0,0,0,4,6],[0,0,0,4,5],[0,0,0,4,4],[0,0,0,4,3],[0,0,0,4,2],[0,0,0,4,1],[0,0,0,4,0]")
                        .map { it.locRound })
                .containsExactly(
                        "Champion",
                        "Runner-up",
                        "Third",
                        "Semifinal",
                        "Quarterfinal",
                        "Group",
                        "Not attended")

        // 12 players in playoffs with three-way final
        Assertions
                .assertThat(UserStatsDto.toDtos("[0,0,1,3,6],[0,0,1,3,5],[0,0,1,3,4],[0,0,1,3,3],[0,0,1,3,2],[0,0,1,3,1],[0,0,1,3,0]")
                        .map { it.locRound })
                .containsExactly(
                        "Champion",
                        "Runner-up",
                        "Third",
                        "Last 6",
                        "Last 12",
                        "Group",
                        "Not attended")

        // 16 players in playoffs
        Assertions
                .assertThat(UserStatsDto.toDtos("[0,0,0,5,7],[0,0,0,5,6],[0,0,0,5,5],[0,0,0,5,4],[0,0,0,5,3],[0,0,0,5,2],[0,0,0,5,1],[0,0,0,5,0]")
                        .map { it.locRound })
                .containsExactly(
                        "Champion",
                        "Runner-up",
                        "Third",
                        "Semifinal",
                        "Quarterfinal",
                        "Last 16",
                        "Group",
                        "Not attended")

        // 24 players in playoffs with three-way final
        Assertions
                .assertThat(UserStatsDto.toDtos("[0,0,1,4,7],[0,0,1,4,6],[0,0,1,4,5],[0,0,1,4,4],[0,0,1,4,3],[0,0,1,4,2],[0,0,1,4,1],[0,0,1,4,0]")
                        .map { it.locRound })
                .containsExactly(
                        "Champion",
                        "Runner-up",
                        "Third",
                        "Last 6",
                        "Last 12",
                        "Last 24",
                        "Group",
                        "Not attended")
    }
}
